smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -57,14 +73,97 @@ public ApplicationProtocol getApplicationProtocol() {
     @Override
     public void generateRequestSerializers(GenerationContext context) {
         TopDownIndex topDownIndex = context.getModel().getKnowledge(TopDownIndex.class);
+        OperationIndex operationIndex = context.getModel().getKnowledge(OperationIndex.class);
+        Walker shapeWalker = new Walker(context.getModel().getKnowledge(NeighborProviderIndex.class).getProvider());
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        TypeScriptWriter writer = context.getWriter();
+
+        // Track the shapes we need to generate sub-serializers for.
+        Set<Shape> serializingShapes = new HashSet<>();
         for (OperationShape operation : topDownIndex.getContainedOperations(context.getService())) {
             OptionalUtils.ifPresentOrElse(
                     operation.getTrait(HttpTrait.class),
                     httpTrait -> generateOperationSerializer(context, operation, httpTrait),
                     () -> LOGGER.warning(String.format(
                             "Unable to generate %s protocol request bindings for %s because it does not have an "
                             + "http binding trait", getName(), operation.getId())));
+
+            operationIndex.getInput(operation)
+                    .ifPresent(input -> serializingShapes.addAll(shapeWalker.walkShapes(input).stream()
+                            // Don't generate a sub-serializer for the actual input shape.
+                            .filter(s -> !input.equals(s))
+                            .collect(Collectors.toSet())));
         }
+
+        // TODO Abstraction point
+        // This should almost certainly be abstracted out to something else, because any protocol
+        // with a document body is going to need to generate sub-serializers.
+        // Generate the serializers for shapes within the operation closure.
+        serializingShapes.forEach(shape -> shape.accept(new ShapeVisitor.Default<Void>() {
+            @Override
+            protected Void getDefault(Shape shape) {
+                return null;
+            }
+
+            private Void generateFunctionSignature(BiConsumer<GenerationContext, Shape> renderer, Shape shape) {
+                Symbol symbol = symbolProvider.toSymbol(shape);
+                // Use the shape name for the function name.
+                String methodName = ProtocolGenerator.getSerFunctionName(symbol, getName());
+                writer.openBlock("const $L = (\n"
+                        + "  input: $T,\n"
+                        + "  context: SerdeContext\n"
+                        + "): any => {", "}", methodName, symbol, () -> {
+                    renderer.accept(context, shape);
+                });
+                writer.write("");
+                return null;
+            }
+
+            @Override
+            public Void listShape(ListShape shape) {
+                // TODO Collection cleanup point
```

Comment:
Is this able to handle lists of lists?

Reply:
Yes:

```
list FooListList {
    member: FooList
}

list FooList {
    member: Foo
}

structure Foo {
   bar: String,
   baz: String
}
```

would generate `serializeAws_restJson1_1FooListList`, `serializeAws_restJson1_1FooList`, and `serializeAws_restJson1_1Foo`.